app/code/Magento/Inventory/Model/Source/Validator/PreventDisableValidator.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Inventory\Model\Source\Validator;
+
+use Magento\Framework\Validation\ValidationResult;
+use Magento\Framework\Validation\ValidationResultFactory;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryCatalogApi\Api\DefaultSourceProviderInterface;
+
+/**
+ * Check that default source always enabled
+ */
+class PreventDisableValidator implements SourceValidatorInterface
```

Comment:
pay attention that SourceValidatorInterface has been moved into another place,
you need to merge 2.3-develop into your branch 
and update SourceValidatorInterface

Reply:
Thanks a lot. I'll update my `2.3-develop` branch and rebase this branch to the current `2.3-develop`.

Reply:
PreventDisableValidator should be moved from `Magento\Inventory` to `Magento\InventoryCatalog` module, because currently we've introduced dependency Inventory -> InventoryCatalogApi which is wrong

Reply:
Yes, I've moved it.